### ISSUE TITLE
Binary Incompatibility between 3.0.1 and (future) 3.1.0

### ISSUE THREAD

**Issue by cortinico** [ChuckerTeam/chucker#199]
## :loudspeaker: Describe the bug
When preparing for the release of 3.1.0, I noticed we might have introduced a couple of breaking changes (both binary and source incompatible changes).

I've run https://github.com/siom79/japicmp and compare `3.0.1` vs the current `develop-SNAPSHOT`.

Here the flags I used (I intentionally excluded the `.internal.` classes):

```
java -jar japicmp-0.14.3-jar-with-dependencies.jar -o library-3.0.1/classes.jar -n library-develop/classes.jar --ignore-missing-classes -a public --html-file report.html -e '*_Impl;*.internal.*' --no-annotations -m
```

Here the report:
[report.html.txt](https://github.com/ChuckerTeam/chucker/files/4094625/report.html.txt)

<img width="608" alt="Screenshot 2020-01-22 at 01 42 44" src="https://user-images.githubusercontent.com/3001957/72855680-88809a80-3cb8-11ea-887d-92b27283e7d4.png">

Here a summary of (minor) problems I found:

* `ChuckerInterceptor`: the change introduced in #189 is source compatible but not binary compatible. This was introduced by @vbuberen. The solution would be to introduce a `@Deprecated fun redactHeader(String)` method. I don't know if it's worth or not in this scenario.

* `RetentionManager` and `ChuckerCollector`: the change introduced in #146 by me, caused the signature of some method to change. E.g. `doMaintenance$library_release` -> `doMaintenance$com_github_ChuckerTeam_Chucker_library()`.
The involved methods are 3: `doMaintenance`, `onResponseReceived` and `onRequestSent`. Those are `internal` method but are actually compiled as `public` method in Java. Hopefully no one was accessing those methods from Java. If so, their code won't compile (this is a source/binary incompatible change). Given that those methods are `internal` in Kotlin, I don't consider them as part of our API surface, so we can probably ignore this problem.

* `ChuckerInterceptorKt`: the class was removed. This class contained only one private field, so I exclude that someone was importing this class in the code.

Long story short: #189 comes with a binary incompatible change. I don't think it's worth to add `@Deprecated` method and I will say that the current `develop` branch is ready to be released as a minor version.

**Comment by vbuberen** [comment]
Thanks for such a detailed comparison.
Regarding my change - I decided to not change function name on purpose, so we won't have breaking changes. As to binary incompatibility - totally agree with you.

**Comment by cortinico** [comment]
Great, so let's close this and we link it in the release notes.
